test/asyncio_tests/test_asyncio_basic.py

```diff
@@ -145,3 +148,28 @@ def __getitem__(self, name):
         coll = db["testcoll"]
         self.assertIsInstance(coll, CollectionSubclass)
         self.assertIsNotNone(await coll.insert_one({}))
+
+
+class ExecutorForkTest(AsyncIOTestCase):
+    @asyncio_test()
+    async def test_executor_reset(self):
+        parent_conn, child_conn = Pipe()
+        init_exec = id(asyncio._EXECUTOR)
+        lock_pid = os.fork()
+
+        if lock_pid == 0:  # Child
+            exec_id = id(asyncio._EXECUTOR)
+            child_conn.send(exec_id)
+            child_conn.send(
+                (
+                    init_exec != exec_id,
+                    "_EXECUTOR was not reinitialized",
+                )
+            )
+            os._exit(0)
+        else:  # Parent
+            self.assertEqual(init_exec, id(asyncio._EXECUTOR))
+            child_exec = parent_conn.recv()
+            self.assertIsNot(child_exec, init_exec)
+            passed, msg = parent_conn.recv()
+            self.assertTrue(passed, msg)
```

Comment:
Instead of testing that the id of asyncio._EXECUTOR changes, we should test that a new motor client can actually be used in the child process.

Reply:
Done.